src/main/frontend/handler/editor.cljs

```diff
@@ -3531,15 +3546,13 @@
 (defn collapse-block! [block-id]
   (when (collapsable? block-id)
     (when-not (skip-collapsing-in-db?)
-      (set-block-property! block-id :collapsed true)
```

Comment:
@tiensonqin Looks like the removal of the set-block-property! calls reintroduces the bug from https://github.com/logseq/logseq/issues/4045#issuecomment-1021564530 . Should we preserve collapse state on blocks?

Reply:
@logseq-cldwalker Thanks for pointing this out! Yes, we should save the collapse state of blocks to the files. I will push a fix later.

Reply:
I just pushed a fix https://github.com/logseq/logseq/commit/4c8d852947c971a040a2850ac7ed7a029b4d8a5a

Some benefits of saving the collapse state in ns `frontend.modules.file.core` instead of using `set-block-property!`:
1. simplify outliner logic, the expand/collapse state is only related to `:block/collapsed`. The later rtc/pure db version doesn't need to deal with `[:block/properties :collapsed]`
2. `save-block-property!` is pretty heavy because it includes datascript transactions, text-transform, and UI refresh.

Reply:
Thanks for the fix and explanation! Confirmed this works 🎉